### PR TITLE
Prevent qualified maps with unresolved issues from getting ranked

### LINKED ISSUE #10654: Prevent qualified maps with open issues/suggestions from getting ranked
While most issues are currently resolved before maps get ranked sometimes a few maps slip through and either never get to apply the suggestion/issue or has to be unranked.

The solution to prevent this from happening is to prevent maps with suggestions/issues from being ranked until the posts have been resolved either by the original poster or an NAT (the NAT will make sure that these posts get resolved)

### PR DESCRIPTION
**Pull request by notbakaneko**
plus an attempt at cleaning up some test methods
closes #10654

- [x] exclude from ranking queue as well?

### CODE REVIEW

**Comment by nanaya** [comment]
Yeah I think it should be excluded from the queue otherwise potentially fewer maps will be ranked at each runs.

**Comment by notbakaneko** [comment]
I tried adding tests for hybrid beatmapsets but I'm not sure the results make sense 🤔 
I wonder if these are related  https://github.com/ppy/osu-web/pull/4911/files#diff-4cf8d05043400424e732b329df53d237b69b3d5c10ad285c894aba37cd4835a4R54 
https://github.com/ppy/osu-web/pull/4489/files#diff-03627ef6eb157031bb0c701c8990ea1c6e7713fc129ef82683b5949bb67ebb39R291-R293

**Comment by nanaya** [comment]
I think the order doesn't matter. By ruleset id:
- 0: anything with 0
- 1: anything with 1 except 0
- 2: anything with 2 except 0 and 1
- 3: anything with 3 and nothing else

**Comment by notbakaneko** [comment]
nevermind, I'm counting them wrong 👀

**Review by nanaya — CHANGES_REQUESTED** [review]

File: app/Console/Commands/ModdingRankCommand.php [hunk 1]
```diff
@@ -25,60 +25,81 @@ class ModdingRankCommand extends Command
...
-        foreach ($modeInts as $modeInt) {
+        foreach ($rulesets as $ruleset) {
             $this->waitRandom();
-            $this->rankAll($modeInt);
+
+            if ($this->countOnly) {
+                $stats = static::getStats($ruleset);
+                $this->info($ruleset->name);
+                foreach ($stats as $key => $value) {
+                    $this->line("{$key}: {$value}");
```
Comment (notbakaneko) [inline app/Console/Commands/ModdingRankCommand.php:73]: would have liked to use `twoColumnDetail` but didn't find any option to make the columns narrower 👀

File: database/factories/BeatmapsetFactory.php [hunk 2]
```diff
@@ -70,9 +71,9 @@ public function pending()
         ]);
     }
 
-    public function qualified()
+    public function qualified(?Carbon $approvedAt = null)
```
Comment (nanaya) [inline database/factories/BeatmapsetFactory.php:74]: `\DateTimeInterface`?

**Review by nanaya — CHANGES_REQUESTED** [review]

**Review by nanaya — APPROVED** [review]

**Pull request merged**